### ISSUE TITLE
Modules hooked to displayNav* are not included on responsive resize event

### ISSUE THREAD

**Issue by roja45** [PrestaShop/PrestaShop#11585]
**Describe the bug**
When the screen is resized an even is emitted and a function is called to move the header components to a mobile view.  This uses the generic _desktop_ selector, however it will only move these components is a target mobile location of the same name already exists.

This mobile section of the header is not accessible to modules however so the mobile target cannot be included.

The module can use javascript to insert the necessary element so that the repsonsive process can run.  On the initial page load this function fires before the module javascript has been loaded, so a module can neither insert the required element, nor receive the emitted event.

How is a module hooked to the displayNav hook supposed to display itself in mobile view?

**To Reproduce**
1. Hook a custom module to displayNav
2. Load the page
3. Resize to mobile
4. Module disappears.

**Screenshots**
If applicable, add screenshots or screenrecords to help explain your problem.

**Additionnal information**
PrestaShop version: 1.7.4.2
PHP version:  7.1

**Comment by khouloudbelguith** [comment]
Hi @roja45,

Thanks for your report.
The default theme is responsive.
So, when you resize the window, the shop should be compatible with the size of this window.
I tried with the module Contact information, it is hooked on displayNav1.
When we resize the window, just the position of this module is changed to be compatible with the size of the window.
I attached a video record.
https://drive.google.com/file/d/1JP_j4B6JLigeXt75rvmy_gTjs1bm2SRs/view
Thanks to check & feedback.

**Comment by roja45** [comment]
Yes, the default modules work fine, but this is because the default theme has been hard-coded to work with them, sorry I didn't make that very clear in the initial report.

```
<div class="hidden-sm-down">
    <div class="col-md-5 col-xs-12">
        {hook h='displayNav1'}
    </div>
    <div class="col-md-7 right-nav">
        {hook h='displayNav2'}
    /div>
</div>
<div class="hidden-md-up text-sm-center mobile">
     <div class="float-xs-left" id="menu-icon">
         <i class="material-icons d-inline">&#xE5D2;</i>
     </div>
     <div class="float-xs-right" id="_mobile_cart"></div>
     <div class="float-xs-right" id="_mobile_user_info"></div>
     <div class="top-logo" id="_mobile_logo"></div>
     <div class="clearfix"></div>
</div>

```

Here you can see a section of the header.tpl. The **hidden-sm-down** section is hidden for small screens, and it is here where the modules are hooked.

The **hidden-md-up** section is displayed for small screens.

There is a javascript function that moves the components from the first section, to the second.  The javascript function uses a generic selector to select the components hooked in the **hidden-sm-down** section, to the **hidden-md-up** section.

The same function however requires that there is an element ready for it in mobile view in order to move it.  You can see **_mobile_cart**, **_mobile_user_info**, and **_mobile_logo** in the **hidden-md-up** section.

This is why your test with contact information worked.

My question with this bug however, is this process cannot work for any other module you want to hook into the displayNav hooks.

**Comment by khouloudbelguith** [comment]
Hi @roja45,

I tried with this module.
[dummymodule.zip](https://github.com/PrestaShop/PrestaShop/files/2639825/dummymodule.zip) & it is displayed when resizing the window.
![image](https://user-images.githubusercontent.com/16067358/49380138-3033b580-f711-11e8-955f-5a2f78af5d56.png)
Thanks to check & feedback.

**Comment by roja45** [comment]
Hi,

Your module is not a valid test, it hooks to displayTop.  I've updated the module to provide an accurate test.

Desktop view - module is shown

![img1](https://user-images.githubusercontent.com/19314911/49383375-9820d680-f6e6-11e8-8649-cc8d7150d8f2.jpg)

![img2](https://user-images.githubusercontent.com/19314911/49383390-9e16b780-f6e6-11e8-906b-34f4fe415fd3.jpg)

Mobile view - module is not shown

[dummymodule.zip](https://github.com/PrestaShop/PrestaShop/files/2640090/dummymodule.zip)

**Comment by khouloudbelguith** [comment]
@roja45, thanks for your feedback.
After your modifications, the module is hooked into the `displayNav2` hook.
When we resize the window or we try with a mobile view, the module is not displayed.
It is not the case with the hook `Top`.
I manage to reproduce the issue with PS1.7.5.0beta1 / 1.7.4.4
We will see how to fix it.
Thanks!

**Comment by marionf** [comment]
@khouloudbelguith Could you try to reproduce it with 1.7.6.0 ?

**Comment by khouloudbelguith** [comment]
Hi @marionf,

Yes, I manage to reproduce the issue with PS1.7.6.0beta1.

When the module is transplanted to the displayNav2 hook, if we resize the window => it is NOT displayed

https://drive.google.com/file/d/1fhLghFeANw6DK1u0QMAqyTODP2_1lhR4/view 

When the module is transplanted to the displayTop hook, if we resize the window => it is well displayed

https://drive.google.com/file/d/1yx3VVpTaV1ovsOl98d8ObMpbaE_JvNtL/view

Thanks!

**Comment by NeOMakinG** [comment]
I'm not able to reproduce it on develop, it looks fine

**Comment by kos2reus** [comment]
@roja45 and @khouloudbelguith you are a serious and professional persons. This issue has not been fixed yet in prestashop 1.7.8 , others commenting on this issue have not tested it properly. Little professionalism in the answers @marionf @NeOMakinG @hibatallahAouadni a disappointment, the problem is not solved yet with displayNav1 hook

**Comment by kos2reus** [comment]
> I'm not able to reproduce it on develop, it looks fine

lie, not checked correctly

**Comment by NeOMakinG** [comment]
> @roja45 and @khouloudbelguith you are a serious and professional persons. This issue has not been fixed yet in prestashop 1.7.8 , others commenting on this issue have not tested it properly. Little professionalism in the answers @marionf @NeOMakinG @hibatallahAouadni a disappointment, the problem is not solved yet with displayNav1 hook

```
Guys, I just tested it and it looks like it's not fixed, maybe @NeOMakinG didn't test it properly, or couldn't reproduce it in his environment, can someone else test again and confirm? So we open the issue again and we can work on fixing it.
```

This is a way better comment and it respects the code of conduct, actually you're being aggressive on humans trying to maintain an open-source project, humans do errors, if you want this bug to get fixed, provide some tests examples, and don't hesitate to contribute to fixing it instead of being aggressive.

@khouloudbelguith, if you want to test it and provide me a way to reproduce it properly if it's the case, feel free to open the issue again.

**Comment by Hlavtox** [comment]
@kos2reus People make mistakes, all the time. If you think the bug is still there and you are professional, you can help to fix it by providing a PR, so you have it integrated in the next release. Sounds good no?

Being a dickhead helps ventilate immediate anger, but does not help anything long term. 👍

**Comment by roja45** [comment]
@Hlavtox @kos2reus not sure what you have against me, I just reported this issue (4 years ago..), and my issue was fixed up nicely by those wonderful peeps at Prestashop.  You two need to get some more zen into your lives.

**Comment by Hlavtox** [comment]
@roja45, sorry mate, wrong tag :D

**Comment by NeOMakinG** [comment]
We tested with @khouloudbelguith 

We were able to reproduce the problem:

- There are no way for modules to add something on mobile, because you've no hook on the mobile part of the header, meaning that you can add something like `<div id="_mobile_dummy_module"></div>` as a placeholder for the mobile version of your module

I think we should introduce a new hook there, wdyt @PrestaShop/prestashop-maintainers @Hlavtox ?

**Comment by matks** [comment]
> I think we should introduce a new hook there

Good idea

**Comment by Hlavtox** [comment]
Reworked for hummingbird completely, not relevant anymore.
